### PR TITLE
Normalize logging

### PR DESCRIPTION
**Pull request by scunningham**
## What is the problem this PR solves?

This PR fixes several issues:
- Another pass at normalizing logging using field names that ElasticSearch will index.
- Uses the proper TLS structure for reading the configuration; thus enabling optional client validation.
- Handles case where enrollment response fails to write, removing orphaned records.
- Enables HTTP logging at the INFO level, adding addition logging if DEBUG enabled.

## How does this PR solve the problem?
// Explain HOW you solved the problem in your code. It is possible that during PR reviews this changes and then this section should be updated.

Refines the logging with constants.  Works the edge cases.

## Checklist

- [x ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: cmd/fleet/handleEnroll.go
Comment (aleksmaus) [inline cmd/fleet/handleEnroll.go:113]: the cleanup will stop if the request is cancelled, this will need to be revisited. agreed to address after this PR is merged.

File: cmd/fleet/handleEnroll.go
Comment (aleksmaus) [inline cmd/fleet/handleEnroll.go:259]: this has a potential to hold the request go routine alive, that might cause the go routines exhaustion at scale.
agreed this will be addressed in the follow up PRs probably as standalone GC routine/pool

**Review by aleksmaus — APPROVED** [review]

File: cmd/fleet/handleEnroll.go
Comment (scunningham) [reply cmd/fleet/handleEnroll.go:259]: Agreed, will address in subsequent PR.

**Comment by jlind23** [comment]
@scunningham could we backport it to 7.15.2?

**Comment by ruflin** [comment]
This was not backported to 8.0. Added the label now. We need to check if other backports are missing too.

**Comment by ruflin** [comment]
This also has not made it into 7.16 or 7.15 as the wrong backport labels were used. Talked to Victor to clean this up.

File: cmd/fleet/handleEnroll.go
Comment (ruflin) [reply cmd/fleet/handleEnroll.go:259]: Is there already a follow up PR for this?

Comment (aleksmaus) [reply cmd/fleet/handleEnroll.go:259]: not yet, was planning to address that

**Comment by scunningham** [comment]
> This also has not made it into 7.16 or 7.15 as the wrong backport labels were used. Talked to Victor to clean this up.
> 
> I don't think we should backport to 7.15, only 7.16 now.

Purposely not back-ported.  Security team gets beat up for landing features post feature freeze. This is not a bugfix or a security issue, so not back-ported.  We can chose back-port anyhow, but I would not do it without good reason.

**Pull request merged**
